src/naming-strategy/NamingStrategyInterface.ts

```diff
@@ -120,5 +120,10 @@ export interface NamingStrategyInterface {
     /**
      * Gets the name of the alias used for relation joins.
      */
-    eagerJoinRelationAlias(alias: string, propertyPath: string): string;
+    joinRelationAlias(alias: string, propertyPath: string, maxAliasLength?: number): string;
+
+    /**
+     * Gets the name of the alias used for eager relation joins.
+     */
+    eagerJoinRelationAlias(alias: string, propertyPath: string, maxAliasLength?: number): string;
```

Comment:
Tbh. I'm not a fan of adding this to NamingStrategy - I think there should be names which are stored in the db, not aliases used in queries(which are runtime only). However `eagerJoinRelationAlias` is there already.

Reply:
I was not a fan either, although since there were one about it already I figured it would be less misleading to have both.
We could split it up in two parts, i.e. the one about the representation in the DB and the one about querying.